mint/git-clone/bin/git-clone

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env bash
+set -ueo pipefail
+
+mkdir -p "${CHECKOUT_PATH}"
+cd "${CHECKOUT_PATH}"
+
+if [[ "${PRESERVE_GIT_DIR}" == "true" ]]; then
+  CREDENTIAL_ARG=""
+  if [[ "${CREDENTIAL_HELPER}" != "" ]]; then
+    CREDENTIAL_ARG="-c credential.helper='${CREDENTIAL_HELPER}'"
+  fi
+  eval "git clone $CREDENTIAL_ARG \"${CHECKOUT_REPOSITORY}\" ."
+  git fetch origin "${CHECKOUT_REF}"
+  git checkout "${CHECKOUT_REF}"
+else
+  git init -b main
+  git remote add origin "${CHECKOUT_REPOSITORY}"
+  if [[ "${CREDENTIAL_HELPER}" != "" ]]; then
+    git config --local credential.helper "${CREDENTIAL_HELPER}"
+  fi
+  git fetch --depth=1 origin "${CHECKOUT_REF}"
+  git checkout FETCH_HEAD
+fi
+
+commit_sha=$(git rev-parse HEAD | tr -d '\n')
+echo "Checked out git repository at ${commit_sha}"
+
+if [[ "${LFS}" == "true" ]]; then
+  LFS_FILES=$(git-lfs ls-files -n)
+  if [[ "${LFS_FILES}" != "" ]]; then
+    echo $LFS_FILES > $MINT_VALUES/lfs-files
+    FILTER_PATH_PREFIX="${CHECKOUT_PATH%/}/"
+    FILTER_LINES=$(echo "$LFS_FILES" | jq -c --raw-input --slurp --arg prefix "$FILTER_PATH_PREFIX" 'split("\n") | map(select(. != "") | $prefix + .)')
+
+    LFS_ENDPOINT=$(git-lfs env | grep "Endpoint=" | awk -F'[ =]' '{print $2}')
+    cat << EOF > $MINT_DYNAMIC_TASKS/lfs.yml
```

Comment:
Another tweak we can probably make later, now, is maybe moving this out to a separate file and using `envsubst`/similar on the specific variables you want to substitute (so you don't have to worry about all of the escaping)